testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -24,612 +23,419 @@ import (
 			pResp, ok := p.(*node.GetIdentityResponse)
 			if !ok {
-				return errJsonCast
-			}
-			lResp, ok := l.(*node.GetIdentityResponse)
-			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &node.GetIdentityResponse{}, p)
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
-			}
-			if lResp.Data == nil {
-				return errEmptyLighthouseData
+				return errEmptyData
 			}
 			return nil
-		},
-	},
-	"/node/peers": {
-		basepath: v1PathTemplate,
-		prysmResps: map[string]interface{}{
-			"json": &node.GetPeersResponse{},
-		},
-		lighthouseResps: map[string]interface{}{
-			"json": &node.GetPeersResponse{},
-		},
-		customEvaluation: func(p interface{}, l interface{}) error {
+		})),
+	"/node/peers": newMetadata[node.GetPeersResponse](v1PathTemplate,
+		withCustomEval(func(p interface{}, _ interface{}) error {
 			pResp, ok := p.(*node.GetPeersResponse)
 			if !ok {
-				return errJsonCast
-			}
-			lResp, ok := l.(*node.GetPeersResponse)
-			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &node.GetPeersResponse{}, p)
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
-			}
-			if lResp.Data == nil {
-				return errEmptyLighthouseData
+				return errEmptyData
 			}
 			return nil
-		},
-	},
-	"/node/peer_count": {
-		basepath: v1PathTemplate,
-		prysmResps: map[string]interface{}{
-			"json": &node.GetPeerCountResponse{},
-		},
-		lighthouseResps: map[string]interface{}{
-			"json": &node.GetPeerCountResponse{},
-		},
-		customEvaluation: func(p interface{}, l interface{}) error {
+		})),
+	"/node/peer_count": newMetadata[node.GetPeerCountResponse](v1PathTemplate,
+		withCustomEval(func(p interface{}, _ interface{}) error {
 			pResp, ok := p.(*node.GetPeerCountResponse)
 			if !ok {
-				return errJsonCast
-			}
-			lResp, ok := l.(*node.GetPeerCountResponse)
-			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &node.GetPeerCountResponse{}, p)
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
-			}
-			if lResp.Data == nil {
-				return errEmptyLighthouseData
+				return errEmptyData
 			}
 			return nil
-		},
-	},
-	"/node/version": {
-		basepath: v1PathTemplate,
-		prysmResps: map[string]interface{}{
-			"json": &node.GetVersionResponse{},
-		},
-		lighthouseResps: map[string]interface{}{
-			"json": &node.GetVersionResponse{},
-		},
-		customEvaluation: func(p interface{}, l interface{}) error {
+		})),
+	"/node/version": newMetadata[node.GetVersionResponse](v1PathTemplate,
+		withCustomEval(func(p interface{}, l interface{}) error {
 			pResp, ok := p.(*node.GetVersionResponse)
 			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &beacon.ListAttestationsResponse{}, p)
 			}
 			lResp, ok := l.(*node.GetVersionResponse)
 			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &beacon.ListAttestationsResponse{}, p)
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
+				return errEmptyData
```

Comment:
Should we specify within the error whether it is `pResp` or `lResp` that has empty data as in this instance if these fail it wont be obvious which of the two did.